modules/processing/autoruns.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import os
+import csv
+
+from lib.cuckoo.common.abstracts import Processing
+from lib.cuckoo.common.exceptions import CuckooProcessingError
+
+log = logging.getLogger(__name__)
+
+__author__ = "[Canadian Centre for Cyber Security] @CybercentreCanada"
+__version__ = "1.0.0"
+
+
+class Autoruns(Processing):
+    def run(self):
+        self.key = "autoruns"
+        autoruns_dir = os.path.join(self.analysis_path, "autoruns")
+        autoruns_data_path = os.path.join(autoruns_dir, "autoruns_autoruns.diff")
+
+        if os.path.exists(autoruns_data_path):
+            autoruns_path = autoruns_data_path
+        else:
+            return
+
+        data = {
+        }
+        try:
+            with open(autoruns_path, "r") as f:
+                #Operation,Time,Entry Location,Entry,Enabled,Category,Profile,Description,Company,Image Path,Version,Launch String
+                reader = csv.DictReader(f, delimiter=",")
+                count = 0
+                for row in reader:
+                    count += 1
+                    data[str(count)] = str(row)
+
+            if count == 0:
+                data = None
```

Comment:
do we want data to be set to None or an empty dict if there is nothing to parse? It feels weird to set it to None if it is initialized to {}.

Reply:
This is what sysmon is doing, I guess it's for preventing empty result.

Reply:
oh, then that should be fixed in sysmon too

Reply:
Will be testing if this is intentionnal i.e required for the processing.